imports/plugins/core/system-info/server/no-meteor/queries/systemInformation.js

```diff
@@ -0,0 +1,21 @@
+import packageJson from "/package.json";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+
+/**
+ * @name queries.systemInformation
+ * @method
+ * @memberof SystemInformation/GraphQL
+ * @summary get systemInformations
+ * @param {String} args.shopId The ID of the shop to load settings for
+ * @param {Object} context - an object containing the per-request state
+ * @return {<Object} System Information
+ **/
+export default async function systemInformation(id, context) {
+  if (!context.userHasPermission(["owner", "admin"], id)) throw new ReactionError("access-denied", "User does not have permission");
+  const plugins = await context.collections.Packages.find({ shopId: id, version: { $exists: true } }).toArray();
```

Comment:
I still think you should replace `{ $exists: true}` with `{ $ne: null }` to avoid any null version packages being included.

Second, stylistically, you should change `id` argument name to `shopId` and change `shopId: id` to just `shopId` in the query.

Also, stylistically, we destructure whenever using keys from an object more than one place in a function, i.e. put this at the top of the function:

```js
const { collections, userHasPermission } = context;
```

In the jsdocs, `args.shopId` should be just `shopId`.

Whenever we pass in `context`, we do it as the first argument, so flip your args. There may be some older functions that don't follow this pattern but we try to always do it now.

Lastly, I don't necessarily agree with restricting this to admins. Currently it is only versions, and this information could be useful to a storefront app, or to anyone who is able to do any queries (which is anyone). If we later add fields that only admins should see, we could implement field-level security at that time.

Reply:
🤦‍♂️ darn i thought i had updated `{ $ne: null }` ! thanks!

I'll remove admin stuff!

Reply:
Could we use a mongo field selection/project to just ask for `name` and `version` from mongodb? I think in general it's good to be precise about that when you know exactly which fields you need, but in this case specifically I think sometimes packages have a black-box `settings` object and in some cases they might be large-ish.

Reply:
On the topic of whether to make this public or restricted, my personal vote is public. My reasoning is it has a high level of utility for troubleshooting, tech support, and general deployment management instrumentation/introspection. 

However, this type of thing is usually disabled in the first round of security hardening. For example, the npm `helmet` security package removes the equivalent header from `express` app servers, and it's common to strip this stuff from Internet-facing web servers. It's a countermeasure to defeat fingerprinting and I think from a security perspective it's a valid practice. Would it make sense to add a configuration setting for this like `[public, admin, disabled]`?

Reply:
Agree with @focusaurus that you should add `{ projection: { name: 1, version: 1 } }` as second arg of the collection `find` call.